config/v1/types_ingress.go

```diff
@@ -82,8 +82,75 @@ type IngressSpec struct {
 	// Note that if there are no RequiredHSTSPolicies, any HSTS Policy annotation on the route is valid.
 	// +optional
 	RequiredHSTSPolicies []RequiredHSTSPolicy `json:"requiredHSTSPolicies,omitempty"`
+
+	// LoadBalancer contains the AWS load balancer details
+	// +optional
+	LoadBalancer LoadBalancer `json:"loadbalancer,omitempty"`
+}
+
+// IngressPlatformSpec holds the desired state of Ingress specific to the underlying infrastructure provider
+// of the current cluster. Since these are used at spec-level for the underlying cluster, it
+// is supposed that only one of the spec structs is set.
+// +union
+type IngressPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
+	// "OpenStack", "VSphere", "oVirt", "KubeVirt", "EquinixMetal", "PowerVS",
+	// "AlibabaCloud", "Nutanix" and "None". Individual components may not support all platforms,
+	// and must handle unrecognized platforms as None if they do not support that platform.
+	//
+	// +unionDiscriminator
+	Type PlatformType `json:"type"`
```

Comment:
This is repeated from the Platform.config.openshift.io so that we can have the discriminated union here?

On upgrade, this will be "", right?  Is "" allowed by validation?

Reply:
If the supported provider is not present in the list of // +kubebuilder:validation:Enum="";AWS;Azure;BareMetal;GCP;Libvirt;OpenStack;None;VSphere;oVirt;IBMCloud;KubeVirt;EquinixMetal;PowerVS;AlibabaCloud;Nutanix
then we shall allow ""

Reply:
@Miciah ^^

Reply:
Right, `PlatformType` allows the empty string: https://github.com/openshift/api/blob/0f86a223d4bce1c63d9adce1f229afb08305a23b/config/v1/types_infrastructure.go#L126-L128

And we are indeed using `PlatformType` so that we can define `spec.loadBalancer.platform` as a discriminated union, making room to expand the API with platform-specific load-balancer settings for other platforms beyond the one option for the one specific platform that Miheer is adding here.